src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ManageContent/KeyStatisticService.cs

```diff
@@ -82,6 +82,31 @@ await _dataBlockService.IsUnattachedDataBlock(releaseId, dataBlock)
             });
     }
 
+    public async Task<Either<ActionResult, KeyStatisticTextViewModel>> CreateKeyStatisticText(
+        Guid releaseId,
+        KeyStatisticTextCreateRequest request)
+    {
+        return await _persistenceHelper.CheckEntityExists<Release>(releaseId)
+            .OnSuccess(_userService.CheckCanUpdateRelease)
+            .OnSuccess(async _ =>
+            {
+                var keyStatisticText = _mapper.Map<KeyStatisticText>(request);
+                keyStatisticText.ReleaseId = releaseId;
+                keyStatisticText.CreatedById = _userService.GetUserId();
+
+                var order = await _context.KeyStatistics
+                    .Where(ks => ks.ReleaseId == releaseId)
+                    .Select(ks => ks.Order)
+                    .MaxAsync(order => (int?)order);
```

Comment:
What's the need for the cast here?

Reply:
This is a hack that allows us to use Max even if it is possible for it to be passed no elements. Without this, if there are no key stats, it throws an `InvalidOperationException`. With this, `var order` is now nullable and so it's fine.